src/ensembl/src/shared/badged-button/BadgedButton.test.tsx

```diff
@@ -50,4 +50,26 @@ describe('BadgedButton', () => {
 
     expect(renderedBadge.hasClass(fakeClassName)).toBe(true);
   });
+
+  it('trims the longer strings to three characters', () => {
+    const wrapper = renderButton(BadgedButton, {
+      badgeContent: 'abcd',
+      children: <Button onClick={onClick}>{faker.lorem.words()}</Button>
+    });
+
+    const renderedBadge = wrapper.find('div.badgeDefault');
```

Comment:
Here and elsewhere: is there any reason to use tag name in the CSS selector? It looks entirely redundant.

Reply:
There will be multiple div elements inside this wrapper but I want to select only one in specific.
Is there a better way to do it?

Reply:
`wrapper.find('.badgeDefault')`?

Reply:
Would it not be better to be a bit more specific by using `div.badgeDefault`?

Reply:
Why?

It's good to be specific when there are different competing elements with the same level of CSS specificity and you need to select a particular one of those. But if we expect that there is only a single element with the class `badgeDefault `, then there is no point in increasing specificity of its CSS selector by adding its tag name.

Reply:
We know that it's just going to be one element and also a div tag. I thought it will make the selection a little bit faster by specifying the tag it needs to look for. I'm aware that there will not be a big difference with just one test, but if we have 50+ tests using CSS selectors, I'm sure it will make the tests to run a little faster.

Reply:
I would be curious to see whether there will be any detectable performance difference at all when using different kinds of CSS selectors during testing; and how many DOM reads there should be in order for the difference to be noticeable. It's the first time I've heard that concern.

In any case, we know that Jest runs its test suites (test files) in parallel.

What I've observed among CSS developers is that they prefer to use only classes by default. Extra selectors are added only when a class is not specific enough to do the job. Don't know if anyone has codified this preference in some kind of a style guide.

Reply:
Ah! This has been reflected in some style guides after all!

Example:

![image](https://user-images.githubusercontent.com/6834224/55727995-2695c380-5a0b-11e9-93ae-d659a74fbfaa.png)
(https://google.github.io/styleguide/htmlcssguide.html#Type_Selectors)

Reply:
Makes more sense that `ul#example` filters `#example` first and then looks for `ul`. In our case, assuming that jest will select `.badgeDefault` at first and then checks if it belongs to a `div`, it would be better to remove the div from the selector to make it faster.